File: Microsoft .NET/.NET Framework 4/ASP.NET/VB.NET/MVC Web Reporting Sample/MVC Web Reporting Sample/DefaultSettings.vb
Imports combit.Reporting
Imports combit.Reporting.DataProviders
Imports combit.Reporting.Repository
Imports combit.Reporting.Web.WebReportDesigner.Server

Namespace WebReporting
    Module CmbtSettings
        ' D:   Stellen Sie hier die Sprache für die Berichte und den Designer ein.
        ' US:  Set the language For the reports And the Designer here.
        Public ReadOnly Property Language As LlLanguage
            Get
                Return LlLanguage.English
                'Return LlLanguage.German
            End Get
        End Property

        ' D: Setzen Sie die gewünschte Einheit auf Inch oder Millimeter. SysDefault Werte verwenden für den WebReportDesigner immer Millimeter, um die Synchronisierung zwischen Client und Server zu gewährleisten.
        ' US: Set the Unit To Inch Or Millimeter. SysDefault Values automatically use millimeter In the web report designer context To make sure client And server units are synchronized. 
        Public ReadOnly Property Unit As LlUnits
            Get
                Return LlUnits.Inch_1_1000
                'Return LlUnits.Millimeter_1_100
            End Get
        End Property
		
        Public ReadOnly Property RepositoryLanguage As String
            Get

                If Language = LlLanguage.German Then
                    Return "de"
                End If

                Return "en"
            End Get
        End Property

        Public ReadOnly Property FileExtension As String
            Get
                If Language = LlLanguage.German Then
                    Return ".lsr"
                End If

                Return ".srt"
            End Get
        End Property
    End Module

    Public Module DefaultSettings
        Private _fileRepository As SQLiteFileRepository

        Public ReadOnly Property DefaultSelectedReport As String
            Get
                If CmbtSettings.Language = LlLanguage.German Then
                    Return "Kundenliste mit Sortierung"
                End If

                Return "Customer list with sort order"
            End Get
        End Property


        Private ReadOnly Property DataMember As String
            Get
                Return "Customers"
            End Get
        End Property

        Function GetBaseRepository() As SQLiteFileRepository
            If _fileRepository Is Nothing Then
                _fileRepository = New SQLiteFileRepository(WebReporting.SampleWebReportingApplication.RepositoryDatabaseFile, CmbtSettings.RepositoryLanguage)
            End If

            Return _fileRepository
        End Function

        Private Function GetDataSourceForProject(ByVal repositoryIdOfProject As String, ByVal forDesign As Boolean) As IDataProvider
            If Not String.IsNullOrEmpty(repositoryIdOfProject) Then
                Dim reportDisplayName As String = GetBaseRepository().GetItem(repositoryIdOfProject).ExtractDisplayName()
                Dim dataProvider As IDataProvider = DataAccess.SampleData.CreateProviderCollection(reportDisplayName & CmbtSettings.FileExtension, forDesign)
                Return dataProvider
            End If

            Return Nothing
        End Function

        Public Function GetDataMemberForProject(ByVal repositoryIdOfProject As String) As String
            If String.IsNullOrEmpty(repositoryIdOfProject) Then
                Return Nothing
            End If

            Dim project As RepositoryItem = GetBaseRepository().GetItem(repositoryIdOfProject)
            If project.Type <> RepositoryItemType.ProjectLabel.Value Then
                Return Nothing
            End If

            Return DataMember
        End Function

        Function GetListLabelInstance(ByVal repositoryID As String, ByVal Optional repository As IRepository = Nothing) As ListLabel
            Dim LL As ListLabel = New ListLabel With {
                .DataSource = GetDataSourceForProject(repositoryID, False),
                .DataMember = GetDataMemberForProject(repositoryID),
                .Language = CmbtSettings.Language,
                .Unit = CmbtSettings.Unit
            }
            'D: Lizenzschlüssel für List & Label setzen. Auf Nicht-Entwicklungsrechnern wird ein Lizenzfehler angezeigt, falls dieser nicht gesetzt wurde.
            'US:  Set license key for List & Label. If not set, a license error will be displayed on non-development machines.
            'LL.LicensingInfo = "<insert your license key here>"
			
            Return LL
        End Function

        Function GetProhibitedActions() As List(Of WebReportDesignerAction)

            'US: You can modify the user prohibitions here by uncommenting actions which should be prohibited.
            'D: Hier können Sie die Benutzerverbote ändern, indem Sie die zu sperrende Aktion einkommentieren.

            Dim prohibited = New List(Of WebReportDesignerAction) From {}
            ' prohibited.Add(WebReportDesignerAction.CreateNewProject)
            ' prohibited.Add(WebReportDesignerAction.BrowseProjects)
            ' prohibited.Add(WebReportDesignerAction.DeleteProject)
            ' prohibited.Add(WebReportDesignerAction.DownloadProject)
            ' prohibited.Add(WebReportDesignerAction.SaveAsProject)
            ' prohibited.Add(WebReportDesignerAction.SaveProject)
            ' prohibited.Add(WebReportDesignerAction.UnlockProject)
            ' prohibited.Add(WebReportDesignerAction.UploadProject)
            Return prohibited
        End Function
    End Module
End Namespace

File: Microsoft .NET/.NET Framework 4/ASP.NET/C#/MVC Web Reporting Sample/MVC Web Reporting Sample/DefaultSettings.cs
using combit.Reporting;
using combit.Reporting.DataProviders;
using combit.Reporting.Repository;
using combit.Reporting.Web.WebReportDesigner.Server;
using System;
using System.Collections.Generic;

namespace WebReporting
{
    static class CmbtSettings
    {
        // D:   Stellen Sie hier die Sprache für die Berichte und den Designer ein.
        // US:  Set the language for the reports and the Designer here.
        public static LlLanguage Language { get { return LlLanguage.English; } }
        //public static LlLanguage Language { get { return LlLanguage.German; } }

        // D: Setzen Sie die gewünschte Einheit auf Inch oder Millimeter. SysDefault Werte verwenden für den WebReportDesigner immer Millimeter, um die Synchronisierung zwischen Client und Server zu gewährleisten.
        // US: Set the Unit to Inch or Millimeter. SysDefault Values automatically use millimeter in the web report designer context to make sure client and server units are synchronized. 
        public static LlUnits Unit { get { return LlUnits.Inch_1_1000; } }
        //public static LlUnits Unit { get { return LlUnits.Millimeter_1_100; } }
		
        public static string RepositoryLanguage
        {
            get
            {
                if (Language == LlLanguage.German)
                {
                    return "de";
                }
                return "en";
            }
        }

        public static string FileExtension
        {
            get
            {
                if (Language == LlLanguage.German)
                {
                    return ".lsr";
                }
                return ".srt";
            }
        }
    }

    public static class DefaultSettings
    {
        private static SQLiteFileRepository _fileRepository;

        public static string DefaultSelectedReport
        {
            get
            {
                if (CmbtSettings.Language == LlLanguage.German)
                {
                    return "Kundenliste mit Sortierung";
                }
                return "Customer list with sort order";
            }
        }

        // D:   Für Etikettenprojekte wird die Tabelle "Customers" ausgewählt.
        // US:  The Table "Customers" is chosen for label projects.
        private static string DataMember
        {
            get
            {
                return "Customers";
            }
        }

        public static SQLiteFileRepository GetBaseRepository()
        {
            if (_fileRepository == null)
            {
                _fileRepository = new SQLiteFileRepository(WebReporting.SampleWebReportingApplication.RepositoryDatabaseFile, CmbtSettings.RepositoryLanguage);
            }

            return _fileRepository;
        }

        // D:   Liefert die passende Datenquelle zu einem Beispiel-Report.
        // US:  Returns the required data source of the sample report.
        private static IDataProvider GetDataSourceForProject(string repositoryIdOfProject, bool forDesign)
        {
            if (!String.IsNullOrEmpty(repositoryIdOfProject))
            {
                // D:   Für dieses Beispielprojekt hängt die benötigte Datenquelle vom Namen des geöffneten Projekt ab, daher muss der Name ausgelesen werden.
                // US:  For this sample project the datasource depends on the name of the opened project, so we need to read the display name.
                string reportDisplayName = GetBaseRepository().GetItem(repositoryIdOfProject).ExtractDisplayName();

                IDataProvider dataProvider = DataAccess.SampleData.CreateProviderCollection(reportDisplayName + CmbtSettings.FileExtension, forDesign);
                return dataProvider;
            }
            return null;
        }

        // D:   Liefert die passende Tabelle zu einem Beispiel-Etikett.
        // US:  Returns the matching data member for a sample label.
        public static string GetDataMemberForProject(string repositoryIdOfProject)
        {
            if (String.IsNullOrEmpty(repositoryIdOfProject))
            {
                return null;
            }

            RepositoryItem project = GetBaseRepository().GetItem(repositoryIdOfProject);

            if (project.Type != RepositoryItemType.ProjectLabel.Value)
            {
                return null;
            }

            return DataMember;
        }

        public static ListLabel GetListLabelInstance(string repositoryID, IRepository repository = null)
        {
            ListLabel LL = new ListLabel
            {

                // D:   Lizenzschlüssel für List & Label setzen. Auf Nicht-Entwicklungsrechnern wird ein Lizenzfehler angezeigt, falls dieser nicht gesetzt wurde.
                // US:  Set license key for List & Label. If not set, a license error will be displayed on non-development machines.
                // LicensingInfo = "<insert your license key here>",

                // D:   Lade die zum Report passende Datenquelle.
                // US:  Get the corresponding data source for the report.
                DataSource = GetDataSourceForProject(repositoryID, false),
                DataMember = GetDataMemberForProject(repositoryID),
                Language = CmbtSettings.Language,
                Unit = CmbtSettings.Unit
            };
			
            return LL;
        }

        public static WebReportDesignerAction[] GetProhibitedActions()
        {
            //US: You can modify the user prohibitions here by uncommenting actions which should be prohibited.
            //D: Hier können Sie die Benutzerverbote ändern, indem Sie die zu sperrende Aktion einkommentieren.
            return new WebReportDesignerAction[]
            {
                //WebReportDesignerAction.CreateNewProject,
                //WebReportDesignerAction.BrowseProjects,
                //WebReportDesignerAction.DeleteProject,
                //WebReportDesignerAction.DownloadProject,
                //WebReportDesignerAction.SaveAsProject,
                //WebReportDesignerAction.SaveProject,
                //WebReportDesignerAction.UnlockProject,
                //WebReportDesignerAction.UploadProject,
            };
        }
    }
}